src/methods/clipping/clipping_processor.jl

```diff
@@ -24,16 +44,20 @@ returns are the fully updated vectors of PolyNodes that represent the rings 'pol
 'poly_b', respectively. This function also returns 'a_idx_list', which at its "ith" index
 stores the index in 'a_list' at which the "ith" intersection point lies.
 =#
-function _build_ab_list(::Type{T}, poly_a, poly_b) where T
+function _build_ab_list(::Type{T}, poly_a, poly_b, delay_cross_f, delay_bounce_f) where T
     # Make a list for nodes of each polygon
     a_list, a_idx_list, n_b_intrs = _build_a_list(T, poly_a, poly_b)
     b_list = _build_b_list(T, a_idx_list, a_list, n_b_intrs, poly_b)
+
     # Flag crossings
     _classify_crossing!(T, a_list, b_list)
 
     # Flag the entry and exits
-    _flag_ent_exit!(GI.LinearRingTrait(), poly_b, a_list)
-    _flag_ent_exit!(GI.LinearRingTrait(), poly_a, b_list)
+    _flag_ent_exit!(GI.LinearRingTrait(), poly_b, a_list, (x) -> delay_cross_f(x), (x) -> delay_bounce_f(x, true))
+    _flag_ent_exit!(GI.LinearRingTrait(), poly_a, b_list, (x) -> delay_cross_f(x), (x) -> delay_bounce_f(x, false))
```

Comment:
I was getting a type instability when I had something like this before, specifically with the first argument. It was just branding it as `Function`. I tried this to make it go away and it worked, but I agree it would be better your way. I can try again and post the profile, but I don't really know how to fix that

Reply:
<img width="1450" alt="Screenshot 2024-03-28 at 5 48 21 PM" src="https://github.com/asinghvi17/GeometryOps.jl/assets/60117338/4e6d3682-7c39-453a-858d-91d955183bb9">

Here is the instability I was talking about. 

<img width="713" alt="Screenshot 2024-03-28 at 5 48 44 PM" src="https://github.com/asinghvi17/GeometryOps.jl/assets/60117338/a1c519bd-5a1f-4500-8305-3b4f0fa29d67">

The top is with the swap removing the anonymous function from the first argument and the second is with the original code.

Reply:
You probably just need `delay_cross_f::F, ... ) where F` on the methods to fix that type instability